tensorrt_llm/_torch/utils.py

```
@@ -275,3 +277,63 @@ def piecewise_cuda_graph(enable: bool):
+    Raises:
+        RuntimeError: If NVML initialization fails
+    """
+    try:
+        pynvml.nvmlInit()
+        yield
+    except pynvml.NVMLError as e:
+        raise RuntimeError(f"Failed to initialize NVML: {e}")
+    finally:
+        try:
+            pynvml.nvmlShutdown()
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)